webapp/tests/karma/ts/modules/reports/reports.component.spec.ts

```diff
@@ -263,4 +279,168 @@ describe('Reports Component', () => {
+          heading: 'report.subject.unknown',
+          icon: undefined,
+          summary: undefined,
+          unread: true,
+        },
+        {
+          _id: 'a86f238a-ad81-4780-9552-c7248864d1b2',
+          lineage:  [ 'Chattanooga Village', 'CHW Bettys Area', null, null ],
+          heading: 'report.subject.unknown',
+          icon: undefined,
+          summary: undefined,
+          unread: true,
+        },
+        {
+          _id: 'd2da792d-e7f1-48b3-8e53-61d331d7e899',
+          lineage: [ 'Chattanooga Village' ],
+          heading: 'report.subject.unknown',
+          icon: undefined,
+          summary: undefined,
+          unread: true,
+        },
+        {
+          _id: 'ee21ea15-1ebb-4d6d-95ea-7073ba357229',
+          lineage: [ 'CHW Bettys Area' ],
+          heading: 'report.subject.unknown',
+          icon: undefined,
+          summary: undefined,
+          unread: true,
+        },
+        {
+          _id: 'ee21ea15-1ebb-4d6d-95ea-7073ba357229',
+          lineage: [],
+          heading: 'report.subject.unknown',
+          icon: undefined,
+          summary: undefined,
+          unread: true,
+        },
+        {
+          _id: 'ee21ea15-1ebb-4d6d-95ea-7073ba965525',
+          lineage: undefined,
+          heading: 'report.subject.unknown',
+          icon: undefined,
+          summary: undefined,
+          unread: true,
+
+        },
+      ];
+
+      component.ngOnInit();
+      component.ngAfterViewInit();
+      flush();
+
+      expect(updateReportsListStub.callCount).to.equal(1);
+      expect(updateReportsListStub.args[0]).to.deep.equal([ expectedReports ]);
+    }));
+
+    it('should remove current level from reports lineage when user is offline', fakeAsync(() => {
+      userContactService.get.resolves(offlineUserContactDoc);
+      sessionService.isOnlineOnly.returns(false);
+      const expectedReports = [
+        {
+          _id: '88b0dfff-4a82-4202-abea-d0cabe5aa9bd',
+          lineage: [ 'St Elmos Concession', 'Chattanooga Village' ],
+          heading: 'report.subject.unknown',
+          icon: undefined,
+          summary: undefined,
+          unread: true,
+        },
+        {
+          _id: 'a86f238a-ad81-4780-9552-c7248864d1b2',
+          lineage:  [ 'Chattanooga Village' ],
+          heading: 'report.subject.unknown',
+          icon: undefined,
+          summary: undefined,
+          unread: true,
+        },
+        {
+          _id: 'd2da792d-e7f1-48b3-8e53-61d331d7e899',
+          lineage: [ 'Chattanooga Village' ],
+          heading: 'report.subject.unknown',
+          icon: undefined,
+          summary: undefined,
+          unread: true,
+        },
+        {
+          _id: 'ee21ea15-1ebb-4d6d-95ea-7073ba357229',
+          lineage: [],
+          heading: 'report.subject.unknown',
+          icon: undefined,
+          summary: undefined,
+          unread: true,
+        },
+        {
```

Comment:
You can delete this object since is the identical to the previous one.

Reply:
it's identical to the previous but I think it's normal, that is the `expectedReports` ie the output after the processing, the input is different.
```
          _id: 'ee21ea15-1ebb-4d6d-95ea-7073ba357229',
          lineage: [ 'CHW Bettys Area' ],
          heading: 'report.subject.unknown',
          icon: undefined,
          summary: undefined,
          unread: true,
        },
        {
          _id: 'ee21ea15-1ebb-4d6d-95ea-7073ba357229',
          lineage: [],
          heading: 'report.subject.unknown',
          icon: undefined,
          summary: undefined,
          unread: true,
        },
```
should become :
```
        {
          _id: 'ee21ea15-1ebb-4d6d-95ea-7073ba357229',
          lineage: [],
          heading: 'report.subject.unknown',
          icon: undefined,
          summary: undefined,
          unread: true,
        },
        {
          _id: 'ee21ea15-1ebb-4d6d-95ea-7073ba357229',
          heading: 'report.subject.unknown',
          icon: undefined,
          lineage: [],
          summary: undefined,
          unread: true,
        },
```
after the content of lineage is updated

Reply:
Ohhh got it, thanks!